aws-crt/spec/spec_helper.rb

```diff
@@ -1,2 +1,16 @@
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
 require 'aws-crt'
+
+# Return whether ObjectSpace.garbage_collect
+# can be relied on to immediately clean up everything possible
+def garbage_collect_is_immediate?
+  RUBY_ENGINE == 'ruby'
+end
+
+def check_for_clean_shutdown
```

Comment:
Naming nitpick: Does this "check" that a clean shutdown has occurred or does it do cleanup?  Ie - if it does check for.... some sort of leaks or uncleaned objects, lets document what exceptions it raises here

Reply:
This just does the checks.
Added comments.

Feel free to rename it, or alter it to fit better into the testing framework in the future.